msal/src/main/java/com/microsoft/identity/client/Account.java

```diff
@@ -93,6 +96,23 @@ public String getId() {
             id = mHomeOid;
         }
 
+        if (StringUtil.isEmpty(id)) {
+            com.microsoft.identity.common.logging.Logger.warn(
+                    TAG + methodName,
+                    "Unable to get account id from either ClientInfo or IdToken. Attempting to obtain from home account id."
+            );
+            id = AccountUtil.getUIdFromHomeAccountId(mHomeAccountId);
+        }
+
+        if (StringUtil.isEmpty(id)) {
+            com.microsoft.identity.common.logging.Logger.warn(
```

Comment:
Should this be an error?

Reply:
probably..

Reply:
I can make that change

Reply:
Addressed https://github.com/AzureAD/microsoft-authentication-library-for-android/pull/1558/commits/c6748c3d7df139933f1629866314a277dca2b1dd